### PR TITLE
Add logic to unpublish rt content

### PR DESCRIPTION
**Pull request by dragospitica**

### CODE REVIEW

File: src/main/java/org/atlasapi/query/v2/ContentWriteController.java [hunk 1]
```diff
@@ -74,6 +79,8 @@
     public static final String ASYNC_PARAMETER = "async";
     public static final String BROADCAST_ASSERTIONS_PARAMETER = "broadcastAssertions";
     private static final String STRICT_PARAMETER = "strict";
+    private static final String SOURCE = "source";
+    private static final String VALID_URIS = "valid_uris";
```
Comment (jamiesprax) [inline src/main/java/org/atlasapi/query/v2/ContentWriteController.java]: I would append these with "_PARAMETER" for consistency

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,357 @@
+FROM 448613307115.dkr.ecr.eu-west-1.amazonaws.com/jvm-oracle:latest
```
Comment (Dalamar42) [inline Dockerfile]: Why are we pushing the dockerfile as part of this change? Those should be unrelated changes.

Stage branch should already have a dockerfile and prod is not on docker yet.

File: src/main/java/org/atlasapi/query/QueryWebModule.java [hunk 3]
```diff
@@ -389,6 +393,14 @@ private ContentWriteController contentWriteController() {
                         contentResolver,
                         entryStore,
                         lookupWriter
+                ),
+                new OldContentDeactivator(
```
Comment (Dalamar42) [inline src/main/java/org/atlasapi/query/QueryWebModule.java]: If any of these classes are part of the `atlas` repo let's change them to have static factory methods as per the code style

File: src/main/java/org/atlasapi/query/v2/ContentWriteController.java [hunk 4]
```diff
@@ -155,9 +167,75 @@ public WriteResponse putContent(HttpServletRequest req, HttpServletResponse resp
...
+
         return setPublishStatus(req, resp, false);
     }
 
+    private WriteResponse unpublishOldContent(
+            HttpServletRequest request,
+            HttpServletResponse response
+    ) {
+        String source = request.getParameter(SOURCE);
+        ImmutableList<String> validUris = ImmutableList.copyOf(
```
Comment (Dalamar42) [inline src/main/java/org/atlasapi/query/v2/ContentWriteController.java]: You can just do 
```
ImmutableList.copyOf(
    request.getParameter(VALID_URIS)
        .split(",")
);
```

File: src/main/java/org/atlasapi/query/v2/ContentWriteController.java [hunk 5]
```diff
@@ -155,9 +167,75 @@ public WriteResponse putContent(HttpServletRequest req, HttpServletResponse resp
...
+            return writeResponse;
+        }
+
+        if (oldContentDeactivator.deactivateOldContent(publisher, validUris, 50)) {
+            response.setStatus(HttpServletResponse.SC_OK);
+        }
+        return null;
+    }
+
+    public WriteResponse validateApiKey(
```
Comment (Dalamar42) [inline src/main/java/org/atlasapi/query/v2/ContentWriteController.java]: This is duplicating logic that already exists in the other methods in this class. Let's split this into two methods and reuse them. The first one will resolve the application configuration and ensure it exists. This one should happen in `#unpublishContent` before anything else.

The second one should validate the application has access to a particular source and should happen for each case where it already is.

File: src/test/java/org/atlasapi/query/v2/ContentWriteControllerTest.java [hunk 6]
```diff
@@ -137,7 +139,7 @@ public void setUp() throws Exception {
         controller = ContentWriteController.create(
```
Comment (Dalamar42) [inline src/test/java/org/atlasapi/query/v2/ContentWriteControllerTest.java]: Let's fix this for the code style. One line per parameter.

File: src/test/java/org/atlasapi/query/v2/ContentWriteControllerTest.java [hunk 7]
```diff
@@ -93,6 +94,7 @@
     private @Mock Identified identified;
     private @Mock Described described;
     private @Mock SubstitutionTableNumberCodec substitutionTableNumberCodec;
+    private @Mock OldContentDeactivator oldContentDeactivator;
```
Comment (Dalamar42) [inline src/test/java/org/atlasapi/query/v2/ContentWriteControllerTest.java]: We need tests for the new logic. At least one that ensures we call the deactivator with the right params from the call and one that ensure that we do not deactivate anything if the application does not have write permissions for the given source.

File: src/test/java/org/atlasapi/query/v2/ContentWriteControllerTest.java [hunk 8]
```diff
@@ -251,4 +274,51 @@ public void postAsynchronously() throws Exception {
...
+        controller.unpublishContent(request, response);
+
+        ArgumentCaptor<AtlasErrorSummary> argument = ArgumentCaptor.forClass(
+                AtlasErrorSummary.class
+        );
+
+        verify(modelWriter, atLeast(2))
+                .writeError(eq(request), eq(response), argument.capture());
+
+        assertEquals("403", argument.getValue().statusCode().toString());
```
Comment (Dalamar42) [inline src/test/java/org/atlasapi/query/v2/ContentWriteControllerTest.java]: Prefer `assertThat(something, is(something));`. It's more readable

File: src/test/java/org/atlasapi/query/v2/ContentWriteControllerTest.java [hunk 9]
```diff
@@ -251,4 +274,51 @@ public void postAsynchronously() throws Exception {
...
+        when(request.getParameter("source")).thenReturn("upcoming.radiotimes.com");
+        when(request.getParameter("valid_uris")).thenReturn("uri1,uri2,uri3");
+
+        controller.unpublishContent(request, response);
+
+        ArgumentCaptor<AtlasErrorSummary> argument = ArgumentCaptor.forClass(
+                AtlasErrorSummary.class
+        );
+
+        verify(modelWriter, atLeast(2))
```
Comment (Dalamar42) [inline src/test/java/org/atlasapi/query/v2/ContentWriteControllerTest.java]: Why are we calling this twice?
